File: app/src/main/java/com/adisa/diningplus/adapters/TraitAdapter.java
package com.adisa.diningplus.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.adisa.diningplus.R;
import com.adisa.diningplus.activities.ItemActivity;

import java.util.ArrayList;

public class TraitAdapter extends BaseAdapter {
    private Context context;
    private ArrayList<ItemActivity.Trait> traits;

    class ViewHolder {
        ImageView image;
        TextView name;
    }

    public TraitAdapter(Context context, ArrayList<ItemActivity.Trait> allergens) {
        this.context = context;
        this.traits = allergens;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        ViewHolder viewHolder;
        ItemActivity.Trait allergen = traits.get(position);
        if (convertView == null) {
            viewHolder = new ViewHolder();
            convertView = inflater.inflate(R.layout.allergen, null);

            viewHolder.image = (ImageView) convertView.findViewById(R.id.trait_img);
            viewHolder.name = (TextView) convertView.findViewById(R.id.trait_text);

            convertView.setEnabled(false);
            convertView.setOnClickListener(null);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        viewHolder.image.setImageDrawable(context.getResources().getDrawable(allergen.image));
        viewHolder.name.setText(allergen.name);

        return convertView;
    }

    @Override
    public int getCount() {
        return traits.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }
}

File: app/src/main/java/com/adisa/diningplus/adapters/TraitAdapter.kt
package com.adisa.diningplus.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.adisa.diningplus.R
import com.adisa.diningplus.activities.ItemActivity.Trait
import java.util.*

class TraitAdapter(private val context: Context, private val traits: ArrayList<Trait>) : BaseAdapter() {

    internal inner class ViewHolder {
        var image: ImageView? = null
        var name: TextView? = null
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var convertView = convertView
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val viewHolder: ViewHolder
        val trait = traits[position]
        if (convertView == null) {
            viewHolder = ViewHolder()
            convertView = inflater.inflate(R.layout.allergen, null)
            viewHolder.image = convertView.findViewById<View>(R.id.trait_img) as ImageView
            viewHolder.name = convertView.findViewById<View>(R.id.trait_text) as TextView
            convertView.isEnabled = false
            convertView.setOnClickListener(null)
            convertView.tag = viewHolder
        } else {
            viewHolder = convertView.tag as ViewHolder
        }
        viewHolder.image!!.setImageDrawable(context.resources.getDrawable(trait.image))
        viewHolder.name!!.text = trait.name
        return convertView!!
    }

    override fun getCount(): Int {
        return traits.size
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

}